client/tokenized-payment-request/tracking.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+import { recordUserEvent } from 'tracks';
+
+let paymentRequestBranding;
+
+// Track the payment request button click event.
+export const trackPaymentRequestButtonClick = ( source ) => {
+	const paymentRequestTypeEvents = {
+		google_pay: 'gpay_button_click',
+		apple_pay: 'applepay_button_click',
+	};
+
+	const event = paymentRequestTypeEvents[ paymentRequestBranding ];
+	if ( ! event ) return;
+
+	recordUserEvent( event, { source } );
+};
+
+// Track the payment request button load event.
+export const trackPaymentRequestButtonLoad = debounce( ( source ) => {
+	const paymentRequestTypeEvents = {
+		google_pay: 'gpay_button_load',
+		apple_pay: 'applepay_button_load',
+	};
+
+	const event = paymentRequestTypeEvents[ paymentRequestBranding ];
+	if ( ! event ) return;
+
+	recordUserEvent( event, { source } );
+}, 1000 );
```

Comment:
```suggestion
// Track the payment request button click event.
export const trackPaymentRequestButtonClick = ( source ) => {
	const paymentRequestTypeEvents = {
		google_pay: 'gpay_button_click',
		apple_pay: 'applepay_button_click',
	};

	const event = paymentRequestTypeEvents[ paymentRequestBranding ];
	if ( ! event ) return;

	recordUserEvent( event, { source } );
};

// Track the payment request button load event.
export const trackPaymentRequestButtonLoad = debounce( trackPaymentRequestButtonClick, 1000 );
```
The bodies of both of these functions appears to be the same. Am I being naïve or could this be refactored to something like the above? 🤔

Reply:
The events are also different (`{gpay/applepay}_button_click` & `{gpay/applepay}_button_load`).

This should allow for the same structure:
```js
const makeTrackingFunction = ( eventName ) => {
	const events = {
		google_pay: `gpay_${ eventName }`,
		apple_pay: `applepay_${ eventName }`,
	};

	return ( source ) => {
		const event = events[ paymentRequestBranding ];
		if ( ! event ) return;

		recordUserEvent( event, { source } );
	};
};

// Track the payment request button click event.
export const trackPaymentRequestButtonClick = makeTrackingFunction(
	'button_click'
);

// Track the payment request button load event.
export const trackPaymentRequestButtonLoad = debounce(
	makeTrackingFunction( 'button_load' ),
	1000
);
```

However, I'm not sure if it's as easy to read.

Reply:
> The events are also different ({gpay/applepay}_button_click & {gpay/applepay}_button_load).

My trusty superpower of not being able to read serves me well once again. 😆 

OK, ignore this suggestion as well then. Feel free to refactor or leave it as is: both options look perfectly valid to me.